### PR TITLE
Fix max lsn bug

### PR DESCRIPTION
**Pull request by jbajic**
## Summary of changes

The problem it fixes is when `request_lsn` is `u64::MAX-1` the `cont_lsn` becomes `u64::MAX` which is the same as `prev_lsn` which stops the loop.

Closes https://github.com/neondatabase/neon/issues/6812

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -2618,18 +2618,21 @@ impl Timeline {
...
-                            key,
-                            Lsn(cont_lsn.0 - 1),
-                            request_lsn,
-                            timeline.ancestor_lsn
-                        ), traversal_path));
+                    if let Some(prev) = prev_lsn {
+                        if prev <= cont_lsn {
+                            // Didn't make any progress in last iteration. Error out to avoid
+                            // getting stuck in the loop.
+                            // We fail here
```
Comment (VladLazar) [inline pageserver/src/tenant/timeline.rs:2625]: Please remove the stray comment.

File: pageserver/src/tenant.rs [hunk 2]
```diff
@@ -4996,4 +4996,24 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_read_at_max_lsn() -> anyhow::Result<()> {
+        //assert!(false);
```
Comment (VladLazar) [inline pageserver/src/tenant.rs:5002]: Please remove this

**Review by VladLazar — COMMENTED** [review]
Hey - Thanks for the contribution! Looks correct to me.
Please remove the stray commented out lines mentioned in the comments.

File: pageserver/src/tenant.rs [hunk 3]
```diff
@@ -4996,4 +4996,23 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_read_at_max_lsn() -> anyhow::Result<()> {
+        let harness = TenantHarness::create("test_get_vectored")?;
```
Comment (VladLazar) [inline pageserver/src/tenant.rs:5002]: This uses the same test directory as `test_get_vectored` and results in some strange CI failures.
Sorry for not spotting it the first time around.

```suggestion
        let harness = TenantHarness::create("test_read_at_max_lsn")?;
```

Comment (jbajic) [reply pageserver/src/tenant.rs:5002]: Aha, I assumed the CI was flaky since the same test passed locally, thanks

**Review by VladLazar — APPROVED** [review]

**Comment by VladLazar** [comment]
@jbajic - just merged this. Thanks for the PR!

**Pull request merged**